test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -831,6 +833,411 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
+			// Check offlined cpus are setting correctly
+			for _, node := range workerRTNodes {
+				offlinedOutput, err := nodes.ExecCommandOnNode([]string{"cat", "/sys/devices/system/cpu/offline"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				offlinedCPUSet, err := cpuset.Parse(offlinedOutput)
+				offlinedCPUSetProfile, err := cpuset.Parse(string(offlinedSet))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(offlinedCPUSet.Equals(offlinedCPUSetProfile))
+			}
+		})
+
+		It("[test_id:50965]Offline Middle CPUID's", func() {
+			var reserved, isolated, offline []string
+			// This map is of the form numaNode[core][cpu-siblings]
+			var numaCoreSiblings map[int]map[int][]int
+			var onlineCPUInt int
+			for _, node := range workerRTNodes {
+				onlineCPUCount, err := nodes.ExecCommandOnNode([]string{"nproc", "--all"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				onlineCPUInt, err = strconv.Atoi(onlineCPUCount)
+				Expect(err).ToNot(HaveOccurred())
+				if onlineCPUInt <= 8 {
+					Skip(fmt.Sprintf("This test needs more than 8 CPUs online to work correctly, current online CPUs are %s", onlineCPUCount))
+				}
+			}
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			for key := range numaCoreSiblings {
+				cores := make([]int, 0)
+				for k := range numaCoreSiblings[key] {
+					cores = append(cores, k)
+				}
+				sort.Ints(cores)
+				middleCoreIds := cores[len(cores)/2]
+				siblings := nodes.GetCpuSiblings(numaCoreSiblings, middleCoreIds)
+				offline = append(offline, siblings...)
+			}
+			offlineCpus := strings.Join(offline, ",")
+			for reservedCores := 0; reservedCores < 2; reservedCores++ {
+				// Get the cpu siblings from the selected core and delete the siblings
+				// from the map. Selected siblings of cores are saved in reservedCpus
+				siblings := nodes.GetCpuSiblings(numaCoreSiblings, reservedCores)
+				reserved = append(reserved, siblings...)
+			}
+			reservedCpus := strings.Join(reserved, ",")
+			// Remaining core siblings available in the
+			// numaCoreSiblings map is used in isolatedCpus
+			for key := range numaCoreSiblings {
+				for k := range numaCoreSiblings[key] {
+					siblings := nodes.GetCpuSiblings(numaCoreSiblings, k)
+					isolated = append(isolated, siblings...)
+				}
+			}
+			isolatedCpus := strings.Join(isolated, ",")
+
+			// Create new performance with offlined
+			reservedSet := performancev2.CPUSet(reservedCpus)
+			isolatedSet := performancev2.CPUSet(isolatedCpus)
+			offlinedSet := performancev2.CPUSet(offlineCpus)
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Enable reserved, isolated and offlined parameters")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reservedSet,
+				Isolated: &isolatedSet,
+				Offlined: &offlinedSet,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
```

Comment:
If we use the same profile that other tests use make sure it reverted to its original value

Reply:
In the BeforeEach section i have the original profile , which i am reverting it back to in AfterEach section